docs/source/gateway/index.rst

```diff
@@ -1,6 +1,971 @@
+    )
+    print(response)
+
+
+Further route types will be added in the future.
+
+MLflow Models
+~~~~~~~~~~~~~
+The ``Mlflow AI Gateway`` is not used exclusively for interactive interaction. It can also be employed as a production service for automated jobs.
+The example below demonstrates how to use an AI Gateway server from within a custom ``pyfunc`` implementation.
+
+.. note::
+    The custom ``Model`` shown in the example below is utilizing environment variables for the AI Gateway server's uri. These values can also be set manually within the
+    definition or can be applied via :func:`mlflow.gateway.get_gateway_uri` after the uri has been set. For the example below, the value for ``MLFLOW_GATEWAY_URI`` is
+    ``http://127.0.0.1:5000/``. For an actual deployment use case, this value would be set to the configured and production deployment server.
+
+.. code-block:: python
+
+    import os
+    import pandas as pd
+    import mlflow
+
+
+    class Model(mlflow.pyfunc.PythonModel):
+        # Define the request mechanism by using the REST API
+        def request(self, route, data):
+            import os
+            import requests
+
+            url = os.environ["MLFLOW_GATEWAY_URI"] + f"gateway/{route}/invocations"
+
+            import json
+
+            response = requests.post(
+                url=url,
+                json=data,
+            )
+            response = json.loads(response.text)
+            return response
+
+        def predict(self, ctx, inp):
+            import os
+            import mlflow
+
+            # Query a particular route for batch operations at a given temperature
+            responses = []
+            for _, row in inp.iterrows():
+                prompt = row["prompt"]
+                response = self.request(
+                    route="completions-claude",
+                    data={
+                        "prompt": prompt,
+                        "temperature": 0.7,
+                    },
+                )
+                responses.append(response["candidates"][0]["text"])
+            return responses
+
+
+    # Log the custom pyfunc model to MLflow
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            python_model=Model(),
+            registered_model_name="anthropic_completions",
+            artifact_path="anthropic_completions",
+        )
+
+    # Use the Model to interface with MLflow AI Gateway for batch inference
+    df = pd.DataFrame.from_dict(
+        {
+            "prompt": [
+                "Tell me about Jupiter",
+                "Tell me about Saturn",
+                "Tell me all about Mars",
+            ]
+        }
+    )
+    print(mlflow.pyfunc.load_model(model_info.model_uri).predict(df))
+
+In practice, an authentication mechanism should be created to control the access of the user (or agent) interfacing with this model to the AI Gateway.
```

Comment:
I think we can remove this. Would be great to tell users that they can evaluate this model with `mlflow.evaluate()`, call it in `mlflow.pyfunc.spark_udf()`, or deploy (link to deployment docs) just like any other MLflow Model :)